src/bucket.ts

```diff
@@ -3174,6 +3185,201 @@ class Bucket extends ServiceObject {
+   *
+   *     - **`publicRead`** - Object owner gets `OWNER` access, and `allUsers`
+   * get `READER` access.
+   * @param {boolean} [options.private] Make the uploaded files private. (Alias for
+   *     `options.predefinedAcl = 'private'`)
+   * @param {boolean} [options.public] Make the uploaded files public. (Alias for
+   *     `options.predefinedAcl = 'publicRead'`)
+   * @param {boolean} [options.resumable] Force a resumable upload. (default:
+   *     true for files larger than 5 MB).
+   * @param {string} [options.uri] The URI for an already-created resumable
+   *     upload. See {@link File#createResumableUpload}.
+   * @param {string} [options.userProject] The ID of the project which will be
+   *     billed for the request.
+   * @param {string|boolean} [options.validation] Possible values: `"md5"`,
+   *     `"crc32c"`, or `false`. By default, data integrity is validated with an
+   *     MD5 checksum for maximum reliability. CRC32c will provide better
+   *     performance with less reliability. You may also choose to skip
+   * validation completely, however this is **not recommended**.
+   */
+  /**
+   * @typedef {array} UploadDirectoryResponse
+   * @property {object} [response] Response for each of the uploaded files.
+   * @param {string} [filename] Absolute file path.
+   * @param {string | Error} [status] Status of the file upload. "success"
+   *     or an Error returned while attempting to upload current file.
+   */
+  /**
+   * @callback UploadDirectoryCallback
+   * @param {?Error} err Request error, if any.
+   * @param {array} resp The list of object responses for every uploaded file.
+   */
+
+  /**
+   * Upload a directory to the bucket. This is a convenience method for multiple
+   * {@link Bucket#upload}.
+   *
+   * You can specify whether or not an upload is resumable by setting
+   * `options.resumable`. *Resumable uploads are enabled by default if your
+   * input file is larger than 5 MB.*
+   *
+   * For faster crc32c computation, you must manually install
+   * [`fast-crc32c`](http://www.gitnpm.com/fast-crc32c):
+   *
+   *     $ npm install --save fast-crc32c
+   *
+   * @see [Upload Options (Simple or Resumable)]{@link https://cloud.google.com/storage/docs/json_api/v1/how-tos/upload#uploads}
+   * @see [Objects: insert API Documentation]{@link https://cloud.google.com/storage/docs/json_api/v1/objects/insert}
+   *
+   * @param {string} directoryPathString The fully qualified path to the
+   *     directory you wish to upload to your bucket.
+   * @param {UploadDirectoryOptions} [options] Configuration options.
+   * @param {UploadDirectoryCallback} [callback] Callback function.
+   * @returns {Promise<UploadDirectoryResponse>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('albums');
+   *
+   * //-
+   * // Upload a directory from a local path.
+   * //-
```

Comment:
The `//-` syntax was for an old version of the docs, we shouldn't be using it anymore. Instead lets use multiple `@example` tags and `<caption>` to add a descriptions to the examples.

Reply:
done